src/main/java/uk/gov/companieshouse/api/accounts/validation/transactionclosure/CurrentPeriodTnClosureValidator.java

```diff
@@ -1,73 +1,51 @@
 package uk.gov.companieshouse.api.accounts.validation.transactionclosure;
 
+import javax.servlet.http.HttpServletRequest;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 import uk.gov.companieshouse.api.accounts.exception.DataException;
-import uk.gov.companieshouse.api.accounts.links.CompanyAccountLinkType;
 import uk.gov.companieshouse.api.accounts.links.SmallFullLinkType;
-import uk.gov.companieshouse.api.accounts.model.rest.CompanyAccount;
 import uk.gov.companieshouse.api.accounts.model.rest.CurrentPeriod;
+import uk.gov.companieshouse.api.accounts.model.rest.SmallFull;
 import uk.gov.companieshouse.api.accounts.model.validation.Errors;
-import uk.gov.companieshouse.api.accounts.service.CompanyAccountService;
 import uk.gov.companieshouse.api.accounts.service.CompanyService;
 import uk.gov.companieshouse.api.accounts.service.impl.CurrentPeriodService;
-import uk.gov.companieshouse.api.accounts.service.impl.SmallFullService;
 import uk.gov.companieshouse.api.accounts.service.response.ResponseObject;
 import uk.gov.companieshouse.api.accounts.service.response.ResponseStatus;
 import uk.gov.companieshouse.api.accounts.validation.BaseValidator;
 
-import javax.servlet.http.HttpServletRequest;
-import java.util.Map;
-
 @Component
 public class CurrentPeriodTnClosureValidator extends BaseValidator {
 
-    private final CompanyAccountService companyAccountService;
-
-    private final SmallFullService smallFullService;
-
     private final CurrentPeriodService currentPeriodService;
 
     private static final String SMALL_FULL_CURRENT_PERIOD_PATH = "$.small_full.current_period";
     private static final String SMALL_FULL_CURRENT_PERIOD_BALANCE_SHEET_PATH = SMALL_FULL_CURRENT_PERIOD_PATH + ".balance_sheet";
 
     @Autowired
     public CurrentPeriodTnClosureValidator(CompanyService companyService,
-                                           CompanyAccountService companyAccountService,
-                                           SmallFullService smallFullService,
                                            CurrentPeriodService currentPeriodService) {
         super(companyService);
-        this.companyAccountService = companyAccountService;
-        this.smallFullService = smallFullService;
         this.currentPeriodService = currentPeriodService;
     }
 
-    public Errors isValid(String companyAccountsId, HttpServletRequest request) throws DataException {
+    public Errors isValid(String companyAccountsId, SmallFull smallFull, HttpServletRequest request,
```

Comment:
Sorry to harp on about method names. But...
isFoo-type methods are by convention boolean. This isn't. Maybe change to validate?

Reply:
Done